Cargo.toml

```diff
@@ -77,6 +77,12 @@ passthrough-decoder = ["librespot-playback/passthrough-decoder"]
 
 default = ["rodio-backend"]
 
+[profile.release-dist-optimized]
+inherits = "release"
+panic = "abort"
+codegen-units = 1
+lto = true
```

Comment:
This and codegen units depend on the linker being used. Unless you intend to pre-package and provide binaries, so you control the full toolchain, I would recommend sticking with Rust defaults and let power users tinker as they want.

Reply:
This is for power users. This does not replace release. To use this you have to pass `--profile release-dist-optimized` when compiling.

Reply:
https://doc.rust-lang.org/rustc/codegen-options/index.html

>This flag controls how many code generation units the crate is split into. It takes an integer greater than 0.

>When a crate is split into multiple codegen units, LLVM is able to process them in parallel. Increasing parallelism may speed up compile times, but may also produce slower code. Setting this to 1 may improve the performance of generated code, but may be slower to compile.

>The default value, if not specified, is 16 for non-incremental builds. For incremental builds the default is 256 which allows caching to be more granular.

The docs say nothing about it being dependent on a linker.

Reply:
LLD is implied as part of LLVM of course. It may be different for mold or some other linker. But anyway it's not a big deal for me.